apps/wpcom-block-editor/src/wpcom/features/redirect-onboarding-user-after-publishing-post.js

```diff
@@ -14,7 +14,17 @@ export function redirectOnboardingUserAfterPublishingPost() {
 	const siteOrigin = getQueryArg( window.location.search, 'origin' );
 	const siteSlug = window.location.hostname;
 
-	dispatch( 'core/edit-post' ).closeGeneralSidebar();
+	const unsubscribeSidebar = subscribe( () => {
+		const isComplementaryAreaVisible = select( 'core/preferences' ).get(
+			'core/edit-post',
+			'isComplementaryAreaVisible'
+		);
+
+		if ( isComplementaryAreaVisible ) {
```

Comment:
While Gutenberg is figuring out if it will open the sidebar or not, we receive `undefined` as return, so checking if it's true will work fine:

<img width="308" alt="image" src="https://user-images.githubusercontent.com/1044309/236004904-79d92bae-4df5-4160-9feb-a880400f4969.png">